src/dispatch/plugins/dispatch_slack/modals/incident/views.py

```diff
@@ -0,0 +1,324 @@
+            description_input_block(initial_value=incident.description),
+            status_select_block(initial_option=incident.status),
+            incident_type_select_block(
+                db_session=db_session,
+                initial_option=incident.incident_type,
+                project_id=incident.project.id,
+            ),
+            incident_priority_select_block(
+                db_session=db_session,
+                initial_option=incident.incident_priority,
+                project_id=incident.project.id,
+            ),
+            tag_multi_select_block(initial_options=incident.tags),
+        ],
+        "close": {"type": "plain_text", "text": "Cancel"},
+        "submit": {"type": "plain_text", "text": "Submit"},
+        "callback_id": UpdateIncidentCallbackId.submit_form,
+        "private_metadata": json.dumps({"channel_id": str(channel_id), "incident_id": incident.id}),
+    }
+
+    return modal_template
+
+
+def report_incident(
+    db_session: SessionLocal,
+    channel_id: str,
+    project_name: str = None,
+    title: str = None,
+    description: str = None,
+):
+    """Builds all blocks required for the reporting incident modal."""
+    project = project_service.get_by_name(db_session=db_session, name=project_name)
+    modal_template = {
+        "type": "modal",
+        "title": {"type": "plain_text", "text": "Incident Report"},
+        "blocks": [
+            {
+                "type": "context",
+                "elements": [
+                    {
+                        "type": "mrkdwn",
+                        "text": "If you suspect an incident and require help, "
+                        "please fill out the following to the best of your abilities.",
+                    }
+                ],
+            },
+            title_input_block(initial_value=title),
+            description_input_block(initial_value=description),
+            project_select_block(db_session=db_session, initial_option=project),
+        ],
+        "close": {"type": "plain_text", "text": "Cancel"},
+        "submit": {"type": "plain_text", "text": "Submit"},
+        "callback_id": ReportIncidentCallbackId.update_view,
+        "private_metadata": json.dumps({"channel_id": str(channel_id)}),
+    }
+
+    # switch from update to submit when we have a project
+    if project:
+        modal_template["callback_id"] = ReportIncidentCallbackId.submit_form
+        modal_template["blocks"] += [
+            incident_type_select_block(db_session=db_session, project_id=project.id),
+            incident_priority_select_block(db_session=db_session, project_id=project.id),
+            tag_multi_select_block(),
+        ]
+
+    return modal_template
+
+
+def update_participant(incident: Incident, participant: Participant = None):
+    """Builds all blocks required for updating the participant modal."""
+    modal_template = {
+        "type": "modal",
+        "title": {"type": "plain_text", "text": "Edit Participant"},
```

Comment:
To be consistent with the command name
```suggestion
        "title": {"type": "plain_text", "text": "Update Participant"},
```